enhancements/dns/plugins.md

```diff
@@ -141,42 +153,53 @@ name queries for our other internal devices using the DNS servers in our data ce
 
 ### Implementation Details/Notes/Constraints
 
-If `ForwardPlugin` consists of more than one `ForwardPlugin`, longest suffix match will be used to determine the
-`ForwardPlugin`. For example, if there are two `ForwardPlugins`, one for foo.com and one for a.foo.com, and the query is
-for www.a.foo.com, it will be routed to the a.foo.com `ForwardPlugin`.
+If `Servers` consists of more than one `Server`, longest suffix match will be used to determine the `Server`. For
+example, if there are two `Servers`, one for "foo.com" and another for "a.foo.com", and the name query is for
+"www.a.foo.com", it will be routed to the `Server` with `Zone` "a.foo.com".
 
-A maximum of 15 `Nameservers` is allowed per `ForwardPlugin`.
+A maximum of 15 `Upstreams` is allowed per `ForwardPlugin`.
 
-The cluster-dns-operator will generate the `forward` plugin configuration of `configmap/dns-default` based on
-`ForwardPlugin` of `dnses.operator.openshift.io` instead of `forward` being statically defined. To achieve this:
+`Name` must comply with the [rfc6335](https://tools.ietf.org/rfc/rfc6335.txt) Service Name Syntax.
+
+`Zones` must conform to the definition of a subdomain in [rf1123](https://tools.ietf.org/html/rfc1123).
+
+The cluster-dns-operator will generate the forward plugin configuration of `configmap/dns-default` based on
+`Servers` of `dnses.operator.openshift.io` instead of the configuration being statically defined. To achieve this:
 
 1. The [default](https://github.com/openshift/cluster-dns-operator/blob/master/assets/dns/configmap.yaml) ConfigMap
 asset and associated code from the [manifests pkg](https://github.com/openshift/cluster-dns-operator/blob/master//pkg/manifests/manifests.go#L70:6)
 should be removed.
 
 2. The [desiredDNSConfigMap](https://github.com/openshift/cluster-dns-operator/blob/master/pkg/operator/controller/controller_dns_configmap.go)
-function must be modified to create a ConfigMap type with a `forward` configuration based on `ForwardPlugin`.
-If `ForwardPlugin` is not present or an invalid `ForwardPlugin` is provided, the ConfigMap will contain
-`forward . /etc/resolv.conf`. Otherwise, desiredDNSConfigMap will use the provided `ForwardPlugin` to construct the
-forwarding configuration and use `/etc/resolv.conf` as a resolver of last resort. For example:
+function must be modified to create a ConfigMap with a server configuration based on `Server`. If `Server` is not
+present or an invalid `Server` is provided, the ConfigMap will not contain the server. If no `Servers` are specified,
+only the default server configuration is used with `forward . /etc/resolv.conf`. Otherwise, desiredDNSConfigMap will use
+the provided `Servers` to construct additional server configuration blocks, each with a forwarding configuration and use
+`/etc/resolv.conf` as a resolver of last resort. For example:
 
 ```yaml
 apiVersion: operator.openshift.io/v1
```

Comment:
Your API group should not be operator.openshift.io/v1.  Is this global config that a customer can set?  If so, config.openshift.io/v1.

Reply:
^ same question being asked here: https://github.com/openshift/enhancements/pull/68/files#diff-cf574d4089fa132eaf6da5ce145261f9R35

Reply:
> Is this global config that a customer can set? If so, config.openshift.io/v1.

@smarterclayton I'm not aware of any component other than cluster-dns-operator that will use the proposed api, so I don't believe it's global. According to [this](https://github.com/openshift/api/blob/master/operator/v1/types_dns.go#L14-L15) it seems like  `operator.openshift.io/v1` is the correct group. Thoughts?

Reply:
The operator DNS CRD is implemented as a singleton, so does that change anything with regards to appropriateness of global config?

If we did use global config, I think we should explicitly document how mutability is handled...

Reply:
@smarterclayton there seems to be confusion remaining here in response to your comments which seem blocking, can you address the followup convo?